core/src/main/scala/org/broadinstitute/dsde/rawls/entities/compact/CompactEntityProvider.scala

```diff
@@ -125,7 +126,23 @@ class CompactEntityProvider(requestArguments: EntityRequestArguments, repository
 
   override def entityTypeMetadata(useCache: Boolean,
                                   parentContext: RawlsRequestContext
-  ): Future[Map[String, EntityTypeMetadata]] = ???
+  ): Future[Map[String, EntityTypeMetadata]] =
+    repository.dataSource.inTransaction(ReadOnly) { _ =>
+      for {
+        entityTypeAndKeys <- repository.queries.listEntityKeys(workspaceId)
+        entityTypeAndCounts <- repository.queries.countEntitiesGroupedByType(workspaceId)
+      } yield {
+        // note that entityTypeAndKeys only contains entity types that have at least one key
+        // and that entityTypeAndCounts contains all entity types, even those with zero keys
+        val keysByType = entityTypeAndKeys.groupMap(_.entityType)(_.attributeKey)
+        entityTypeAndCounts.map { case EntityTypeAndCount(entityType, count) =>
+          entityType -> EntityTypeMetadata(count,
+                                           entityType + Attributable.entityIdAttributeSuffix,
+                                           keysByType.getOrElse(entityType, Seq.empty)
```

Comment:
for functional equivalence with the current implementation, I think this needs to be
```scala
keysByType.getOrElse(entityType, Seq.empty).sortBy(_.toLowerCase)
```
to alphabetize the attribute names. You could potentially move that sorting into an `ORDER BY` in the sql; not sure if that is worth it